contracts/verifier/BoltzDeployVerifier.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier:MIT
+// solhint-disable no-inline-assembly
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "../TokenHandler.sol";
+import "../factory/BoltzSmartWalletFactory.sol";
+import "../interfaces/IDeployVerifier.sol";
+import "../interfaces/EnvelopingTypes.sol";
+import "../interfaces/BoltzVerifier.sol";
+
+/**
+ * A Verifier to be used on deploys.
+ */
+contract BoltzDeployVerifier is IDeployVerifier, TokenHandler {
+    address private _factory;
+
+    constructor(address walletFactory) public {
+        _factory = walletFactory;
+    }
+
+    function versionVerifier()
+        external
+        view
+        virtual
+        override
+        returns (string memory)
+    {
+        return "rif.enveloping.token.iverifier@2.0.1";
+    }
+
+    /* solhint-disable no-unused-vars */
+    function verifyRelayedCall(
+        EnvelopingTypes.DeployRequest calldata relayRequest,
+        bytes calldata signature
+    ) external virtual override returns (bytes memory context) {
+        require(
+            relayRequest.relayData.callForwarder == _factory,
+            "Invalid factory"
+        );
+
+        address contractAddr = BoltzSmartWalletFactory(
+            relayRequest.relayData.callForwarder
+        ).getSmartWalletAddress(
+                relayRequest.request.from,
+                relayRequest.request.recoverer,
+                relayRequest.request.index
+            );
+
+        require(!_isContract(contractAddr), "Address already created!");
+
+        if (relayRequest.request.tokenAmount > 0) {
+            if (relayRequest.request.tokenContract != address(0)) {
+                require(
+                    tokens[relayRequest.request.tokenContract],
```

Comment:
Same comment as before, in the minimalistic scenario, Boltz would not use ERC20 tokens